src/Contract/Operation/Distinctable.php

```diff
@@ -19,8 +19,10 @@
 interface Distinctable
 {
     /**
-     * @param null|callable(mixed): (Closure(mixed): bool) $comparatorCallback
-     * @param null|callable(T, TKey): mixed $accessorCallback
+     * @template U
+     *
+     * @param null|callable(U): (Closure(U): bool) $comparatorCallback
+     * @param null|callable(T, TKey): U $accessorCallback
```

Comment:
note a very important element: the type returned by the accessor callback needs to match the type of the parameters in the closures from the comparator callback, otherwise a type error will happen at runtime